nominal/core.py

```diff
@@ -311,6 +339,132 @@ def _from_conjure(cls, client: NominalClient, dataset: scout_catalog.EnrichedDat
+            next_page_token=response.next_page_token,
+            client=self._client,
+        )
+
+    @classmethod
+    def _from_conjure(cls, client: NominalClient, log_set_metadata: datasource_logset_api.LogSetMetadata) -> Self:
+        timestamp_type = None
+        if log_set_metadata.timestamp_type == datasource.TimestampType.ABSOLUTE:
+            timestamp_type_enum = "absolute"
+        elif timestamp_type_enum == datasource.TimestampType.RELATIVE:
+            timestamp_type_enum = "relative"
+        else:
+            raise ValueError(f"unsupported timestamp type: {timestamp_type}")
+
+        return cls(
+            rid=log_set_metadata.rid,
+            created_by=log_set_metadata.created_by,
+            name=log_set_metadata.name,
+            created_at=parser.parse(log_set_metadata.created_at),
+            updated_at=parser.parse(log_set_metadata.updated_at),
+            log_count=log_set_metadata.log_count,
+            timestamp_type=timestamp_type,
+            origin_metadata=MappingProxyType(log_set_metadata.origin_metadata),
+            description=log_set_metadata.description,
+            _client=client,
+        )
+
+
+@dataclass(frozen=True)
+class Log:
+    time: datetime | IntegralNanosecondsUTC
+    body: str
+    properties: Mapping[str, str] | None = None
+
+    def _to_conjure(self) -> datasource_logset_api.Log:
+        properties = {} if not self.properties else dict(self.properties)
+        return datasource_logset_api.Log(
+            time=_flexible_time_to_global_conjure_api(self.time),
+            body=datasource_logset_api.LogBody(
+                basic=datasource_logset_api.BasicLogBody(
+                    properties=properties,
+                    message=self.body,
+                ),
+            ),
+        )
+
+    @classmethod
+    def _from_conjure(cls, log: datasource_logset_api.Log) -> Self:
+        return cls(
+            time=_global_conjure_api_to_integral_nanoseconds(log.time),
+            body=log.body.basic.message,
+            properties=MappingProxyType(log.body.basic.properties),
+        )
+
+
+@dataclass(frozen=True)
+class PaginatedLogs:
+    logs: Sequence[Log]
+    page_size: int
+    next_page_token: str | None = None
+    _client: NominalClient = field(repr=False)
+
+    def get_logs(self) -> Sequence[Log]:
+        return self.logs
+
+    def load_next_page(self) -> PaginatedLogs | None:
+        request = datasource_logset_api.SearchLogsRequest(
+            page_size=self.page_size,
+            page_token=self.next_page_token,
+        )
+        response = self._client._logset_client.search_logs(
+            self._client._auth_header,
+            log_set_rid=self.rid,
+            request=request,
+        )
+        return PaginatedLogs(
+            logs=[Log._from_conjure(log) for log in response.logs],
+            page_size=self.page_size,
+            next_page_token=response.next_page_token,
+            client=self._client,
+        )
```

Comment:
don't love that the user would have to call `load_next_page()` and then `get_logs()`, open to ideas of how to deal with pagination cleanly.